nmigen/build/run.py

```diff
@@ -162,3 +224,24 @@ def get(self, filename, mode="b"):
         super().get(filename, mode)
         with open(os.path.join(self.__root, filename), "r" + mode) as f:
             return f.read()
+
+
+class RemoteSSHBuildProducts(BuildProducts):
+    def __init__(self, connect_to, root):
+        self.__connect_to = connect_to
+        self.__root = root
+
+    def get(self, filename, mode="b"):
+        super().get(filename, mode)
+
+        from paramiko import SSHClient
+
+        with SSHClient() as client:
+            client.load_system_host_keys()
+            client.connect(**self.__connect_to)
+
+            with client.open_sftp() as sftp:
+                sftp.chdir(self.__root)
+
+                with sftp.file(filename, "r" + mode) as f:
```

Comment:
The `mode` is [ignored](http://docs.paramiko.org/en/stable/api/sftp.html#paramiko.sftp_client.SFTPClient.open) here, so this likely breaks any downstream code that expects text.

Reply:
What do you mean by "downstream" code? Should I be converting to *nix line endings when writing the file (i.e. iterate over lines)?

Reply:
Well, if something calls `extract` and passes `"t"` as a mode it expects a `str` back, not `bytes`, and will break since this code returns `bytes`.